src/program.cpp

```diff
@@ -2243,8 +2252,19 @@ bool cvk_entry_point::allocate_descriptor_sets(VkDescriptorSet* ds) {
                                             &descriptorSetAllocateInfo, ds);
 
     if (res != VK_SUCCESS) {
-        cvk_error_fn("could not allocate descriptor sets: %s",
-                     vulkan_error_string(res));
+        if (config.enqueue_command_retry_sleep_us == UINT32_MAX) {
+            cvk_error_fn("could not allocate descriptor sets: %s",
+                         vulkan_error_string(res));
+        } else if (m_first_allocation_failure) {
+            cvk_warn_fn(
+                "could not allocate descriptor sets: %s, retry in %u us",
+                vulkan_error_string(res),
+                config.enqueue_command_retry_sleep_us());
+        } else {
+            cvk_info_fn("could not allocate descriptor sets: %s",
+                        vulkan_error_string(res));
+        }
+        m_first_allocation_failure = false;
```

Comment:
It's quite a bit of complexity just to improve log messages :). Fine with it but I note the following:
- Maybe we should start thinking of a `cvk_warn_once` function that'd encapsulate this.
- The message about retry could get out of sync since the decision is made much higher up in the call stack. Ideally the message would be printed by the bit of code that handles the retry.